posts/2022/08-27-speed-up-ci-migrations.md

```diff
@@ -0,0 +1,212 @@
+==title==
+Speed up Elixir CI Migrations with Database dumps
+
+==author==
+Alan Vardy
+
+==footer==
+
+==description==
+How to speed up PostgreSQL migrations on Continuous Integration services (and locally) using database structure dumps
+
+==tags==
+elixir ecto postgres
+
+==body==
+
+## The problem we are solving
+
+Every time the test suite runs on a continuous integration service like CircleCI or GitHub Actions, the database usually needs to be created and migrated from scratch. This doesn't take long when the application is young and not many migrations have been created, but as the migrations folder grows so will the time that developers need to wait. Eventually, that wait becomes long enough (and running `mix ecto.reset` locally becomes painful enough) that utilizing database structure dumps becomes very attractive.
+
+As an example, at the current company I work with, we utilize multiple databases with many tables and migrating all of them locally (on a beefy Threadripper processor) takes 38 seconds:
+
+```bash
+> time mix ecto.reset
+
+Executed in  141.59 secs    fish           external
+   usr time    2.85 secs  600.00 micros    2.85 secs
+   sys time    0.57 secs  145.00 micros    0.57 secs
+```
+
+Whereas using the dump takes only 38 seconds (I have overridden reset in my `mix.exs`)
+
+```bash
+> time mix ecto.reset
+
+Executed in   38.27 secs    fish           external
+   usr time    2.50 secs    0.00 micros    2.50 secs
+   sys time    0.38 secs  712.00 micros    0.38 secs
+```
+
+This can be a huge difference when we are talking about this executing on **every single CI run**.
```

Comment:
There is a lot of "this" going on. Could it be re-phrased more elegantly?
This makes a huge difference, especially considering that this  - or a better word - is executed on ...

Reply:
Removed a ton of "this"